go/vt/discovery/tablet_health_check.go

```diff
@@ -200,7 +200,7 @@ func (thc *tabletHealthCheck) processResponse(hc *HealthCheckImpl, shr *query.St
 	thc.setServingState(serving, reason)
 
 	// notify downstream for master change
-	hc.updateHealth(thc.SimpleCopy(), prevTarget, trivialUpdate, true)
+	hc.updateHealth(thc.SimpleCopy(), prevTarget, trivialUpdate, thc.Serving)
```

Comment:
nit: `thc.Serving` should not be false in practice (?) might want to add a warning log here when we change master to a tablet that is still down.

Reply:
It can be true or false, and this gets called for all tablet types. It is just that we only do anything with that value if the tablet_type is MASTER.
The `healthy` list update code inside the call to `updateHealth` is pretty solid, I think. It handles the true/false cases correctly. It will never replace with a `down` master.
Does this address your concern?

Reply:
ah got it. so this code actually "incorrectly" always claim `isPrimaryUp=true` in health update
this looks good